src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,45 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  padding: 20px;
+  width: $calendar-width;
+  gap: $box-gap;
+
+  @each $day, $number in $days {
+    &--start-day--#{$day} &__day:first-child {
+      margin-inline-start: ($box-width + 3 * $box-gap) * ($number - 2);
+    }
+  }
+
+  @for $amount from 28 through 31 {
+    &--month-length--#{$amount} {
+      :nth-child(n + #{$amount + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: $box-width;
+    height: $box-width;
+    border: 1px solid #000;
+    border-radius: 5px;
+    background-color: $box-color;
+
+    @for $i from 1 through 31 {
+      &:nth-child(#{$i})::before {
+        content: "#{$i}";
+      }
+    }
+
+    &:hover {
+      background-color: $color-secondary;
+      cursor: pointer;
+      transform: translateY(-20px);
+      transition-duration: 0.5s;
```

Comment:
the transition should be set for all the states
otherwise background-color disappears immediately